python/res/simulator/simulation_context.py

```diff
@@ -123,6 +115,8 @@ def get_run_context(self):
 
     def stop(self):
         self._queue_manager.stop_queue( )
+        if self.sim_thread is not None:
```

Comment:
Is this necessary? Should not `self.sim_thread` always be a `Thread`?

Reply:
ok, that was the legacy from the previous version where `self.sim_thread` was initialized independently. fixed now